### ISSUE TITLE
Error in X %*% Z : non-conformable arguments

### ISSUE THREAD

**Issue by garyzhubc** [weizhouUMICH/SAIGE#296]
Hi, Does anyone knows what this error means?

```
[1] "OK1"
     user    system   elapsed 
93635.937 82065.840  8436.268 
[1] 397
geneID:  CRNN 
[1] "CRNN\trs182719074\trs41267182\trs200186996\trs191663957\trs184974054\trs80175531\trs750428070\trs556038275\trs541620206\trs552707325\trs564853736\trs539329993\trs562997842\trs746283392"
genetic variants with  2.703287e-05 <= MAF <=  1 are included for gene-based tests
[1] "ids_to_include"
 [1] "rs182719074" "rs41267182"  "rs200186996" "rs191663957" "rs184974054"
 [6] "rs80175531"  "rs750428070" "rs556038275" "rs541620206" "rs552707325"
[11] "rs564853736" "rs539329993" "rs562997842" "rs746283392"
TEST 14
TEST 1 OK
TEST2 14
ranges_to_include.nrow() 0
ranges_to_exclude.nrow() 0
ids_to_include.size() 14
ids_to_exclude.size() 0
14 markers will be analyzed 
          used  (Mb) gc trigger  (Mb)  max used   (Mb)
Ncells 2029718 108.4    5568117 297.4   3536750  188.9
Vcells 8279882  63.2   30589753 233.4 214999143 1640.4
          used  (Mb) gc trigger  (Mb)  max used   (Mb)
Ncells 2029708 108.4    5568117 297.4   3536750  188.9
Vcells 8279840  63.2   30589753 233.4 214999143 1640.4
Mtest:  14 
          used  (Mb) gc trigger  (Mb)  max used   (Mb)
Ncells 2029718 108.4    5568117 297.4   3536750  188.9
Vcells 8279840  63.2   30589753 233.4 214999143 1640.4
closed the genofile!
cntMarker:  14 
isCondition is  FALSE 
weights:  0.1889508 0.1888824 0.1896651 0.1881275 0.1902716 0.188833 0.1892468 0.18842 0.1922952 0.1895669 0.1884658 0.1882637 0.1910288 0.1878918 
$p.value
[1] 1

$param
$param$p.val.each
[1] 0.8601882 0.8601144 0.8599689 0.8598427 0.8596984 0.8596332 0.8595936

$param$q.val.each
[1]  43.33874  48.85835  65.41716  93.01518 181.32885 319.31896 594.74722

$param$rho
[1] 0.00 0.01 0.04 0.09 0.25 0.50 1.00

$param$minp
[1] 0.8595936

$param$rho_est
[1] 1


$p.value.resampling
NULL

$IsMeta
[1] TRUE

$markerNumbyMAC
[1]  0  0  0  0  0  0  0 14

$m
[1] 14

$indexNeg
integer(0)

time for SAIGE_SKAT_withRatioVec
   user  system elapsed 
  9.240   8.127   0.382 
saigeskatTest$p.value:  1 
[1] "OK1"
     user    system   elapsed 
93655.875 82083.324  8439.775 
[1] 398
geneID:  CROCC 
[1] "CROCC\trs530021629\trs570276740\trs191479229\trs377726134\trs11587728\trs61461842\trs115862221\trs532828390\t1:17251231_CA_C\trs547590389\trs574142808\trs533464043\trs138609368\t1:17252040_AG_A\trs567518338\trs78293422\trs34438479\trs546478535\trs539386940\trs577308231\trs146652978\trs548966643\trs116462658\trs373961830\trs137905506\trs199891773\trs376960739\trs536344053\trs370846905\trs560322539\trs573888531\trs114418071\t1:17259156_AGGT_A\trs555052458\trs191111857\trs567371221\trs80043975\trs541778273\trs189337269\trs116114310\trs537066663\trs564347355\trs370142824\trs533471615\trs6586575\trs149317645\trs367858382\trs45452999\trs762748855\trs546376253\trs563072088\trs138240666\trs142266983\trs532963325\trs552000602\trs202137386\trs6684687\trs6676927\trs193024541\trs7528179\trs7412801\trs375726926\trs548244433\trs183582181\trs549027353\trs533844297\trs553963304\trs183814717\trs190027705\trs181805316\trs192222309\trs185393745\trs575643367\trs149945465\trs556538065\trs566194777\trs569365221\trs558900241\trs572902226\trs572965201\trs116621884\trs186487371\trs111979748\trs536622853\trs41272733\trs74875673\trs541543104\trs41272741\t1:17281560_CAAG_C\trs190663661\trs558376449\trs567017914\trs147351556\trs183726251\trs187875383\trs566901115\trs556184853\trs572016373\trs185387089\trs189497557\trs12085826\trs72899415\trs76576326\trs554800183\trs542390338\trs182425530\trs12142109\t1:17288090_TG_T\trs531826222\trs187290869\trs557731522\trs6691985\trs536289659\trs527911941\trs56729297\t1:17290811_TTC_T\trs572248289\trs148743305\trs187575878\trs191905976\trs79286051\trs538759802\trs139131241\trs149111636\trs72639732\trs570681947\trs41272743\trs559228097\trs115823338\trs35991555\trs77226181\trs187126457\trs74058342\trs373566591\trs139786167\trs146260866\trs79354809\trs145138931\trs147926226\trs9435656\trs200022049\trs145314784\trs115155267\trs59768494\trs185815555\trs199842653\trs536043306\trs6661019"
genetic variants with  2.703287e-05 <= MAF <=  1 are included for gene-based tests
[1] "ids_to_include"
  [1] "rs530021629"       "rs570276740"       "rs191479229"      
  [4] "rs377726134"       "rs11587728"        "rs61461842"       
  [7] "rs115862221"       "rs532828390"       "1:17251231_CA_C"  
 [10] "rs547590389"       "rs574142808"       "rs533464043"      
 [13] "rs138609368"       "1:17252040_AG_A"   "rs567518338"      
 [16] "rs78293422"        "rs34438479"        "rs546478535"      
 [19] "rs539386940"       "rs577308231"       "rs146652978"      
 [22] "rs548966643"       "rs116462658"       "rs373961830"      
 [25] "rs137905506"       "rs199891773"       "rs376960739"      
 [28] "rs536344053"       "rs370846905"       "rs560322539"      
 [31] "rs573888531"       "rs114418071"       "1:17259156_AGGT_A"
 [34] "rs555052458"       "rs191111857"       "rs567371221"      
 [37] "rs80043975"        "rs541778273"       "rs189337269"      
 [40] "rs116114310"       "rs537066663"       "rs564347355"      
 [43] "rs370142824"       "rs533471615"       "rs6586575"        
 [46] "rs149317645"       "rs367858382"       "rs45452999"       
 [49] "rs762748855"       "rs546376253"       "rs563072088"      
 [52] "rs138240666"       "rs142266983"       "rs532963325"      
 [55] "rs552000602"       "rs202137386"       "rs6684687"        
 [58] "rs6676927"         "rs193024541"       "rs7528179"        
 [61] "rs7412801"         "rs375726926"       "rs548244433"      
 [64] "rs183582181"       "rs549027353"       "rs533844297"      
 [67] "rs553963304"       "rs183814717"       "rs190027705"      
 [70] "rs181805316"       "rs192222309"       "rs185393745"      
 [73] "rs575643367"       "rs149945465"       "rs556538065"      
 [76] "rs566194777"       "rs569365221"       "rs558900241"      
 [79] "rs572902226"       "rs572965201"       "rs116621884"      
 [82] "rs186487371"       "rs111979748"       "rs536622853"      
 [85] "rs41272733"        "rs74875673"        "rs541543104"      
 [88] "rs41272741"        "1:17281560_CAAG_C" "rs190663661"      
 [91] "rs558376449"       "rs567017914"       "rs147351556"      
 [94] "rs183726251"       "rs187875383"       "rs566901115"      
 [97] "rs556184853"       "rs572016373"       "rs185387089"      
[100] "rs189497557"       "rs12085826"        "rs72899415"       
[103] "rs76576326"        "rs554800183"       "rs542390338"      
[106] "rs182425530"       "rs12142109"        "1:17288090_TG_T"  
[109] "rs531826222"       "rs187290869"       "rs557731522"      
[112] "rs6691985"         "rs536289659"       "rs527911941"      
[115] "rs56729297"        "1:17290811_TTC_T"  "rs572248289"      
[118] "rs148743305"       "rs187575878"       "rs191905976"      
[121] "rs79286051"        "rs538759802"       "rs139131241"      
[124] "rs149111636"       "rs72639732"        "rs570681947"      
[127] "rs41272743"        "rs559228097"       "rs115823338"      
[130] "rs35991555"        "rs77226181"        "rs187126457"      
[133] "rs74058342"        "rs373566591"       "rs139786167"      
[136] "rs146260866"       "rs79354809"        "rs145138931"      
[139] "rs147926226"       "rs9435656"         "rs200022049"      
[142] "rs145314784"       "rs115155267"       "rs59768494"       
[145] "rs185815555"       "rs199842653"       "rs536043306"      
[148] "rs6661019"        
TEST 148
TEST 1 OK
TEST2 148
ranges_to_include.nrow() 0
ranges_to_exclude.nrow() 0
ids_to_include.size() 148
ids_to_exclude.size() 0
148 markers will be analyzed 
          used  (Mb) gc trigger  (Mb)  max used   (Mb)
Ncells 2029712 108.4    5568117 297.4   3536750  188.9
Vcells 8244416  62.9   30589753 233.4 214999143 1640.4
          used  (Mb) gc trigger  (Mb)  max used   (Mb)
Ncells 2029702 108.4    5568117 297.4   3536750  188.9
Vcells 8244109  62.9   30589753 233.4 214999143 1640.4
Mtest:  148 
          used  (Mb) gc trigger  (Mb)  max used   (Mb)
Ncells 2029712 108.4    5568117 297.4   3536750  188.9
Vcells 8244109  62.9   30589753 233.4 214999143 1640.4
closed the genofile!
cntMarker:  148 
isCondition is  FALSE 
weights:  0.188547 0.1883624 0.1896911 0.1917327 0.217381 0.2470007 0.1884658 0.1892692 0.1897503 0.1886293 0.1901981 0.1924907 0.1910657 0.1894606 0.1881827 0.2548888 0.1890533 0.191111 0.1879164 0.1886058 0.1928281 0.1908562 0.1894204 0.1888718 0.1881933 0.189156 0.190089 0.1884212 0.1902348 0.1893177 0.1897491 0.2470007 0.1921407 0.1949967 0.1897881 0.1882121 0.1889968 0.1876739 0.1896166 0.1896757 0.1896154 0.1891265 0.188893 0.1886399 24.96758 0.1908788 0.1883283 0.2013843 0.1887753 0.189405 0.1893554 0.1889378 0.2624394 0.1883107 0.1893424 0.1899563 0.1896107 0.1919934 0.1898757 0.1876095 0.1896592 0.1911062 0.1888447 0.1892185 0.1877488 0.1883612 0.1913005 0.1898272 0.1888341 0.1906374 0.1911324 0.1887529 0.1970832 0.1879094 0.1889685 0.1882719 0.1891843 0.1912147 0.1895315 0.1900451 0.1890887 0.1921479 0.2015648 0.1907111 0.1960245 0.1894736 0.1882943 0.1920569 0.1890097 0.1883025 0.1885129 0.1902562 0.188607 0.1886517 0.1907099 0.1884835 0.1898899 0.1887105 0.1880032 0.1881909 0.1895587 0.1888506 0.2650269 0.1885705 0.1883189 0.1917638 0.2021084 0.1988491 0.1886952 0.190223 0.1891076 0.8123208 0.1901494 0.1885258 0.1900641 0.1886634 0.1890828 0.1882097 0.1943624 0.190306 0.1875861 0.1905162 0.1914867 0.1886834 0.1884388 0.1876657 0.1918642 0.1882896 0.1973545 0.2180695 0.1909169 0.1910098 0.2717808 0.1886552 0.1954832 0.1893672 0.1884658 0.1884658 0.1889366 7.198703 0.1924019 0.194469 0.2140586 0.1945042 0.19278 0.1908003 0.1900499 23.22652 
$p.value
[1] 0.6302179

$param
$param$p.val.each
[1] 0.4687092 0.5707355 0.6705526 0.6952361 0.7293956 0.7400139 0.7454581

$param$q.val.each
[1]  44364.36  46654.35  53524.32  64974.26 101614.07 158863.78 273134.19

$param$rho
[1] 0.00 0.01 0.04 0.09 0.25 0.50 1.00

$param$minp
[1] 0.4687092

$param$rho_est
[1] 0


$p.value.resampling
NULL

$IsMeta
[1] TRUE

$markerNumbyMAC
[1]   1   0   0   0   0   0   0 147

$m
[1] 148

$indexNeg
integer(0)

time for SAIGE_SKAT_withRatioVec
   user  system elapsed 
140.671 109.066  25.641 
saigeskatTest$p.value:  0.6302179 
[1] "OK1"
Error in X %*% Z : non-conformable arguments
Calls: SPAGMMATtest ... groupTest -> scoreTest_SAIGE_quantitativeTrait_sparseSigma
Execution halted
Done
```

**Comment by cxfustc** [comment]
In 1865th line in file 'SAIGE_SPATest.R', I think 'Z = A1 * g1' should be 'Z = t(A1) * g1'.

**Comment by weizhouUMICH** [comment]
Sorry fo the late reply! We have just released a new version 1.0.0. It has computational efficiency improvements for both Step 1 and Step 2 for single-variant and set-based tests. We have created a new program github page https://github.com/saigegit/SAIGE with the documentation provided https://saigegit.github.io/SAIGE-doc/
The program will be maintained by multiple SAIGE developers there. The docker image has been updated. Please feel free to try the version 1.0.0 and report issues if any.

Thanks!
Wei
